File: Data Structures & Algorithms/count-paths/submission-2.java
class Solution {
    public int uniquePaths(int m, int n) {
      int d = m - 1;
      int r = n - 1;
      int k = Math.min(d, r);
      int N = d + r;
      long res = 1L;
      for (int i = 1; i <= k; i++) {
        res = res * (N- k + i) / i;
      }
      return (int) res;
    }
}

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-2.py
class Solution:
    def minCostClimbingStairs(self, cost):
        n = len(cost)
        prev2, prev1 = 0, 0
        for i in range(2, n + 1):
            from_prev = prev1 + cost[i - 1]
            from_prev2 = prev2 + cost[i - 2]
            curr = min(from_prev, from_prev2)
            prev2, prev1 = prev1, curr
        return prev1